### PR TITLE
chore: Use released lumina-node instead of vendoring it

### PR DESCRIPTION
**Pull request by fl0rek**

### CODE REVIEW

File: src/parts/Form/index.jsx [hunk 1]
```diff
@@ -189,7 +168,7 @@ const Form = () => {
     // NOTE • Start the node
     const startNode = async () => {
         if (!bootnodes || bootnodes.length === 0) {
-            alert('Genesis hash and at least one bootnode are required.');
+            console.error('Genesis hash and at least one bootnode are required.');
```
Comment (zvolin) [inline src/parts/Form/index.jsx:171]: I think alert should stay, as it presents error to the user

File: src/parts/Lumina/index.jsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { useRef, useState, useEffect, createContext } from "react";
+
+import { spawnNode } from "lumina-node";
+
+export const LuminaContext = createContext(null);
+
+export function LuminaContextProvider({ children }) {
+  const [lumina, setLumina] = useState(null);
+  const initialized = useRef(false);
```
Comment (zvolin) [inline src/parts/Lumina/index.jsx:9]: can we switch to `useState` here? I overused useRef I think :)

File: src/parts/Form/index.jsx [hunk 3]
```diff
@@ -63,16 +64,11 @@ const Form = () => {
 
     // NOTE • Refs are needed to access state from within events.onmessage callbacks
     const statsRef = useRef(stats);
-    const nodeRef = useRef(node);
 
     useEffect(() => {
         statsRef.current = stats;
     }, [stats]);
 
-    useEffect(() => {
```
Comment (zvolin) [inline src/parts/Form/index.jsx:72]: as we discussed async, I think nodeRef should be restored

File: src/parts/Form/index.jsx [hunk 4]
```diff
@@ -3,9 +3,10 @@
 // Imports
 // ------------
 import React, { useState, useEffect, useRef, useContext, useLayoutEffect } from 'react';
-import init, { Network, NodeClient, NodeConfig } from '@public/lumina-node-wasm';
+import { spawnNode, Network, NodeClient, NodeConfig } from 'lumina-node';
```
Comment (zvolin) [inline src/parts/Form/index.jsx:6]:
```suggestion
import { Network, NodeConfig } from 'lumina-node';
```

File: src/parts/Lumina/index.jsx (hunk 2 above)
Comment (fl0rek) [reply src/parts/Lumina/index.jsx:9]: I'm not sure how to make it not initialise worker twice (that's what's happening with naively replacing ref with state here) :sweat_smile: 
I'll tinker a bit more

File: src/parts/Form/index.jsx (hunk 3 above)
Comment (fl0rek) [reply src/parts/Form/index.jsx:72]: I think I thought of another way - passing `node` from context to `startNode` that's being called in [this useEffect](https://github.com/eigerco/lumina-front/pull/55/files#diff-612d149cf44513fa470680443f89faf2b3f9604e61ba936a83b75a1ee2cd7fddR305). As I understand it it should result in updated node instance getting passed. lmk what you think.

File: src/parts/Form/index.jsx (hunk 1 above)
Comment (zvolin) [reply src/parts/Form/index.jsx]: I mean, likely they did, but console error is not a best replacement, it doesn't give any feedback for a user without a console open

**Review by zvolin — APPROVED** [review]
nice :+1:

**Pull request merged**